scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -269,6 +272,24 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
                             "and $scannerCriteria, not scanning the package again $packageIndex."
                 }
 
+                if (createMissingArchives) {
+                    val packagePath = pkg.id.toPath()
+                    val missingArchives = HashSet<Provenance>()
+
+                    storedResults.forEach {
+                        if (!archiver.hasArchive("$packagePath/${it.provenance.hash()}")) {
+                            missingArchives.add(it.provenance)
+                        }
+                    }
```

Comment:
Could be written as

                    val missingArchives = storedResults.mapNotNullTo(mutableSetOf()) { result ->
                        result.provenance.takeUnless { archiver.hasArchive("$packagePath/${result.provenance.hash()}") }
                    }